src/AutoRest.CSharp.V3/ClientModels/ClientBuilder.cs

```diff
@@ -185,16 +182,41 @@ private static ServiceClientParameter[] OrderParameters(IEnumerable<ServiceClien
             );
 
             string operationName = operation.CSharpName();
+            //TODO: This is a hack since we don't have the model information at this point
+            var schemaForPaging = ((responseBody as ObjectResponseBody)?.Type as SchemaTypeReference)?.Schema as ObjectSchema;
             return new ClientMethod(
                 operationName,
                 ClientModelBuilderHelpers.EscapeXmlDescription(operation.Language.Default.Description),
                 request,
                 OrderParameters(methodParameters),
                 clientResponse,
-                new ClientMethodDiagnostics($"{clientName}.{operationName}", Array.Empty<DiagnosticScopeAttributes>())
+                new ClientMethodDiagnostics($"{clientName}.{operationName}",Array.Empty<DiagnosticScopeAttributes>()),
+                GetClientMethodPaging(operation, schemaForPaging)
             );
         }
 
+        private static ClientMethodPaging? GetClientMethodPaging(Operation operation, ObjectSchema? schema)
+        {
+            var pageable = operation.Extensions.GetValue<IDictionary<object, object>>("x-ms-pageable");
+            if (pageable == null) return null;
+
+            var nextLinkName = pageable.GetValue<string>("nextLinkName");
+            //TODO: This should actually reference an operation: https://github.com/Azure/autorest.csharp/issues/397
+            var operationName = pageable.GetValue<string>("operationName");
+
+            var itemName = pageable.GetValue<string>("itemName");
+            //TODO: Hack to figure out the property name on the model
+            var itemProperty = schema?.Properties?.FirstOrDefault(p => p.SerializedName == itemName);
+            itemName = itemProperty?.CSharpName() ?? itemName ?? "Value";
```

Comment:
Is Value an appropriate default?

Reply:
Yes. I actually have tests for it. 'value' is the default when itemName is unspecified. So, our property would get cased as 'Value'.